numba/core/types/abstract.py

```diff
@@ -111,16 +116,16 @@ def mangling_args(self):
         """
         return self.name, ()
 
-    def __repr__(self):
+    def __repr__(self) -> str:
         return self.name
 
-    def __hash__(self):
+    def __hash__(self) -> int:
         return hash(self.key)
 
-    def __eq__(self, other):
+    def __eq__(self, other: pt.Any) -> bool:
```

Comment:
since the code attempts to get the `.key` property, only `NumbaTypeInst` are allowed. Would it be valid and more precise to use `NumbaTypeInst` instead of `pt.Any`?

Reply:
Done.